### PR TITLE
Add a `filterByType` method to TraversableLikes

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

**Review by danbills — APPROVED** [review]
cool!

File: src/main/scala/lenthall/collections/EnhancedCollections.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package lenthall.collections
+
+import scala.collection.TraversableLike
+import scala.collection.generic.CanBuildFrom
+import scala.reflect.ClassTag
+
+object EnhancedCollections {
+
+  /**
+    * After trying and failing to do this myself, I got this to work by copying the answer from here:
```
Comment (mcovarr) [inline src/main/scala/lenthall/collections/EnhancedCollections.scala:10]: 😆

**Review by mcovarr — APPROVED** [review]
:+1: assuming there's nothing here that would ruffle wdl4s cross compilation to 2.11, which I don't *think* there is

**Comment by mcovarr** [comment]
:+1: 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/lenthall/pull-request/37/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/lenthall)

**Pull request merged**